File: recette/src/app/models/baseCategorie.ts

export class BaseCategorie {
    id: number;
    nom: string;
    ordre: number;
    isDefaultCategory: boolean = false;


    constructor(defaultValues = false) {
        if (defaultValues) {
            this.nom = "Aucune catégorie"
            this.ordre = 0;
            this.id = 0;
            this.isDefaultCategory = true;
        }
    }

    equals(other: BaseCategorie): boolean {
        return this.id === other.id && this.nom === other.nom && this.ordre === other.ordre && this.isDefaultCategory === other.isDefaultCategory;
    }
}

File: recetteApi/RecetteApi/RecetteClassLibrary/Models/BaseCategorie.cs
namespace RecetteClassLibrary.Models
{
    public abstract class BaseCategorie
    {
        public int Id { get; set; }

        public string? Nom { get; set; }

        public int Ordre { get; set; }

        public bool IsDefaultCategory { get; set; }

        public object toDbModel(int recetteId)
        {
            return new
            {
                this.Nom,
                this.Ordre,
                this.IsDefaultCategory,
                Recette_Id = recetteId
            };
        }

    }
}
